### PR TITLE
feat(middleware): Require workspace for /api/private calls

### PR DESCRIPTION
**Pull request by martwozniak**
Now all requests send to `/api/private/*` require passing `workspace` in url or in cookie.

### CODE REVIEW

File: middleware.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { NextRequest, NextResponse } from "next/server"
+
+export { default } from "next-auth/middleware"
+
+export function middleware(request: NextRequest) {
+
+    if(request.nextUrl.searchParams.get('workspace') || request.cookies.has('workspace')){
```
Comment (grski) [inline middleware.ts:7]: searchParams searches through the url elements or kwargs/args?

File: middleware.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        const response = NextResponse.next();
+        return response;
+    } else {
+        return NextResponse.json({error: 404},  {
+            status: 404,
```
Comment (grski) [inline middleware.ts:12]: 401/403 check which is better in such case :)

File: middleware.ts (hunk 1 above)
Comment (martwozniak) [reply middleware.ts:7]: The searchParams readonly property of the [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL) interface returns a [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) object allowing access to the [GET](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/GET) decoded query arguments contained in the URL.

Source: https://developer.mozilla.org/en-US/docs/Web/API/URL/searchParams

Comment (martwozniak) [reply middleware.ts:7]: Example:

```
  // URL -> `/api/private/test?workspace=default`
  // `workspace` -> 'default'
```

File: middleware.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+export { default } from "next-auth/middleware"
+
+export function middleware(request: NextRequest) {
+
+    if(request.nextUrl.searchParams.get('workspace') || request.cookies.has('workspace')){
+        const response = NextResponse.next();
+        return response;
+    } else {
+        return NextResponse.json({error: 401},  {
```
Comment (grski) [inline middleware.ts:11]: can we get the statuses abstracted away as to not use magic numbers?
eg
statuses.HTTP_401_UNAUHTORIZED = 401

then in the code
eeror: statuses.HTTP_401_UNATHOURIZED

**Review by grski — CHANGES_REQUESTED** [review]
check out the comment

**Comment by martwozniak** [comment]
Updated

**Review by grski — APPROVED** [review]

**Pull request merged**